packages/devtools_shared/lib/src/deeplink/deeplink_manager.dart

```diff
@@ -54,4 +112,69 @@ class DeeplinkManager {
       kOutputJsonField: outputJson,
     };
   }
+
+  Future<Map<String, Object?>> getAndroidAppLinkSettings({
+    required String rootPath,
+    required String buildVariant,
+  }) {
+    return _runFlutterCommandAndReadOutputFile(
+      <String>[
+        'analyze',
+        '--android',
+        '--output-app-link-settings',
+        '--build-variant=$buildVariant',
+        rootPath,
+      ],
+    );
+  }
+
+  Future<Map<String, Object?>> getIosBuildOptions({
```

Comment:
this method can use `_runFlutterCommandAndReadOutputFile` if we add two parameters `regex` and `void Function(ProcessResult) noMatchError`

```dart
return _runFlutterCommandAndReadOutputFile(
  ['analyze', '--ios', '--list-build-options', rootPath],
  regex: _iosBuildOptionsJsonRegex,
  (result) => 'Can not parse output ${result.stdout}',
);
```